platform/core-api/core-api.iml

```diff
@@ -22,5 +22,6 @@
     </orderEntry>
     <orderEntry type="library" name="asm" level="project" />
     <orderEntry type="library" name="CGLIB" level="project" />
+    <orderEntry type="library" name="Guava" level="project" />
```

Comment:
I certainly can! But is Guava a heavyweight dependency? I find it's one of those libraries I use everywhere. Then again I work at Google so maybe that's not surprising ;)

Anyway, without Guava I can't use ListenableFutures, which are an arguably much more ergonomic way to use Futures. Eg. Futures.allAsList(futures). But yeah I could avoid it at the cost of reimplementing the bits that I need. Up to you!